### PR TITLE
Gatsby 5 + ESM compatibility

### PR DESCRIPTION
**Pull request by zbynek**
* peer dependency bumped to avoid warnings with Gatsby 5
* validation changed for ESM to work, analogous to https://github.com/gatsbyjs/gatsby/commit/9dcc77340675a3b9a2f57c82e4b4e19c57072cfb#diff-81450433be21e6b741ce2d6bfeec00ad357ef3cb9126b2a5119797b776f0ee05L21

To verify that these changes are sufficient for ESM support see build status of: https://github.com/jenkins-infra/plugin-site/pull/1476

### CODE REVIEW

**Comment by halkeye** [comment]
do you want to clone these into the repo instead? I'll still merge and figure out release, but it doesn't seem useful enough for others for releases

File: packages/gatsby-transformer-rehype/src/gatsby-node.js [hunk 1]
```diff
@@ -12,18 +12,7 @@ exports.pluginOptionsSchema = function({Joi}) {
...
-                Joi.array()
-                    .items(
-                        Joi.string(),
-                        Joi.object({
-                          resolve: Joi.string(),
-                          options: Joi.object({}).unknown(true),
-                        })
-                    )
-                    .default([]),
+    plugins: Joi.subPlugins(),
```
Comment (halkeye) [inline packages/gatsby-transformer-rehype/src/gatsby-node.js:15]: did something break? or just cleanup?

File: packages/gatsby-transformer-rehype/package.json [hunk 2]
```diff
@@ -46,6 +46,6 @@
         "cross-env": "7.0.3"
     },
     "peerDependencies": {
-        "gatsby": "^3.0.0 || ^4.0.0"
+        "gatsby": "^4.0.0 || ^5.0.0"
```
Comment (halkeye) [inline packages/gatsby-transformer-rehype/package.json:49]: is there a reason to drop 3 support?

File: packages/gatsby-transformer-rehype/src/gatsby-node.js (hunk 1 above)
Comment (zbynek) [reply packages/gatsby-transformer-rehype/src/gatsby-node.js:15]: The validation fails if you pass `{entry: 'index'}` for ESM-bases sub-plugin (stacktrace https://ci.jenkins.io/job/Infra/job/plugin-site/job/PR-1476/9/console). Without that it should work for both Common JS and ESM.

File: packages/gatsby-transformer-rehype/package.json (hunk 2 above)
Comment (zbynek) [reply packages/gatsby-transformer-rehype/package.json:49]: If we want the short validation without "if subplugins defined", we need https://github.com/gatsbyjs/gatsby/commit/3260b1a43ace66349b0a683ab052f0d5c10f33d2 which is part of 4.0.

**Comment by zbynek** [comment]
@halkeye if plugin site is the only consumer of this we can add it to the repo 🤷 It would just be nice to avoid having two diverging versions.

**Comment by halkeye** [comment]
i'm not aware of anyone else using it. I just been distracted and havn't had a chance to merge and remember how to release. I'll try to do it after work.

**Comment by halkeye** [comment]
+ @halkeye/gatsby-rehype-autolink-headers@1.1.0
+ @halkeye/gatsby-rehype-rewrite-ids@1.1.0
+ @halkeye/gatsby-transformer-rehype@1.1.0

🙏  🤞

**Comment by zbynek** [comment]
Thanks! The `unist-*` packages could now be updated.

**Pull request merged**